test/unit/active_model/serializer/associations_test.rb

```diff
@@ -14,6 +14,36 @@ def test_associations_inheritance
         assert_equal([:comments],
                      another_inherited_serializer_klass._associations.keys)
       end
+      def test_multiple_nested_associations
+        parent = SelfReferencingUserParent.new(name: "The Parent")
```

Comment:
@groyoh I have just tried the same thing but I get the error:

1) Error:
ActiveModel::Serializer::AssociationsTest#test_multiple_nested_associations:
NoMethodError: undefined method `type' for #<SelfReferencingUser:0x82859058>
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:126:in `block (2 levels) in associate'
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:252:in `serialize_ids'
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:172:in `block in associations'
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:169:in `each'
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:169:in `each_with_object'
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:169:in `associations'
    /home/webcomputing/active_model_serializers/lib/active_model/serializer.rb:293:in `serializable_object'
    /home/webcomputing/active_model_serializers/lib/active_model/serializable.rb:12:in `block in as_json'
    /home/webcomputing/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activesupport-4.2.6/lib/active_support/notifications.rb:166:in `instrument'
    /home/webcomputing/active_model_serializers/lib/active_model/serializable.rb:55:in `instrument'
    /home/webcomputing/active_model_serializers/lib/active_model/serializable.rb:10:in `as_json'
    /home/webcomputing/active_model_serializers/test/unit/active_model/serializer/associations_test.rb:21:in `test_multiple_nested_associations'